File: examples/1st-level/sum-of-a-list.py
# function:
# =========
# name: sumList
# parameters:
#   - numbers: list of numbers
# return type: number

def sum_list(numbers):
    running_total = 0
    
    for n in numbers:
        running_total += n # or running_total = running_total + n
    
    return running_total

# test cases
print("sum_list([3, 10, 1]):", sum_list([3, 10, 1]), "should be 14")
print("sum_list([3, 10, 1, -4]):", sum_list([3, 10, 1, -4]), "should be 10")
print("sum_list([0, 0, 0, 1]):", sum_list([0, 0, 0, 1]), "should be 1")
print("sum_list([0, 0]):", sum_list([0, 0]), "should be 0")

File: examples/1st-level/sum-of-a-list.js
// function:
// =========
// name: sumList
// parameters:
//  - numbers: list of numbers
// return type: number

function sumList(numbers) {
  let runningTotal = 0;

  for (let n of numbers) {
    runningTotal += n; // or runningTotal = runningTotal + n;
  }

  return runningTotal;
}

// test cases
console.log("sumList([3, 10, 1]):", sumList([3, 10, 1]), "should be 14");
console.log("sumList([3, 10, 1, -4]):", sumList([3, 10, 1, -4]), "should be 10");
console.log("sumList([0, 0, 0, 1]):", sumList([0, 0, 0, 1]), "should be 1");
console.log("sumList([0, 0]):", sumList([0, 0]), "should be 0");
